### PR TITLE
[bazel] update dvsim.py to build ROMs with bazel

### LINKED ISSUE #11562: [bazel] update DV sim to build ROMs with bazel
To run DV simulations with a ROM image built with Bazel, some modifications must be made:

- [x] update the dvsim.py companion makefile (sim.mk) to invoke bazel instead of meson for ROM images
- [x] update the top-level DV testbench to load ROM images from the bazel build directory

### PR DESCRIPTION
**Pull request by timothytrippel**
This fixes #11562 by update dvsim.py to build ROM images with bazel.
Note, flash images are still built with meson, and will be transitioned
to bazel in a future PR.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

**Review by weicaiyang — APPROVED** [review]
LGTM

**Review by cindychip — COMMENTED** [review]
Thanks for the PR, looks like CI failure in top-level sw tests:

                [139/139] Generating plic_all_irqs_test_export_sim_dv with a custom command
                Building SW image "sw/device/lib/testing/test_rom/test_rom".
                Building "//sw/device/lib/testing/test_rom" with bazel on network-connected machine.
                /bin/bash: line 24: bazel: command not found
                /azp/agent/_work/1/s/hw/dv/tools/dvsim/sim.mk:58: recipe for target 'sw_build' failed
                make: *** [sw_build] Error 127

    * Test chip_sw_pwrmgr_smoketest has 1 failures.
        * 0.chip_sw_pwrmgr_smoketest.1\
         Log /azp/agent/_work/1/s/scratch/HEAD/chip_earlgrey_asic-sim-vcs/0.chip_sw_pwrmgr_smoketest/out/run.log

                [17/21] Generating pwrmgr_smoketest_vmem64_sim_dv with a custom command
                [18/21] Generating pwrmgr_smoketest_sim_dv_logs with a custom command
                [19/21] Generating pwrmgr_smoketest_dis_sim_dv with a custom command (wrapped by meson to capture output)
                [20/21] Generating pwrmgr_smoketest_scr_vmem64_sim_dv with a custom command
                [21/21] Generating pwrmgr_smoketest_export_sim_dv with a custom command
                Building SW image "sw/device/lib/testing/test_rom/test_rom".
                Building "//sw/device/lib/testing/test_rom" with bazel on network-connected machine.
                /bin/bash: line 24: bazel: command not found
                /azp/agent/_work/1/s/hw/dv/tools/dvsim/sim.mk:58: recipe for target 'sw_build' failed
                make: *** [sw_build] Error 127

**Comment by timothytrippel** [comment]
> Thanks for the PR, looks like CI failure in top-level sw tests:
> 
> ```
>             [139/139] Generating plic_all_irqs_test_export_sim_dv with a custom command
>             Building SW image "sw/device/lib/testing/test_rom/test_rom".
>             Building "//sw/device/lib/testing/test_rom" with bazel on network-connected machine.
>             /bin/bash: line 24: bazel: command not found
>             /azp/agent/_work/1/s/hw/dv/tools/dvsim/sim.mk:58: recipe for target 'sw_build' failed
>             make: *** [sw_build] Error 127
> 
> * Test chip_sw_pwrmgr_smoketest has 1 failures.
>     * 0.chip_sw_pwrmgr_smoketest.1\
>      Log /azp/agent/_work/1/s/scratch/HEAD/chip_earlgrey_asic-sim-vcs/0.chip_sw_pwrmgr_smoketest/out/run.log
> 
>             [17/21] Generating pwrmgr_smoketest_vmem64_sim_dv with a custom command
>             [18/21] Generating pwrmgr_smoketest_sim_dv_logs with a custom command
>             [19/21] Generating pwrmgr_smoketest_dis_sim_dv with a custom command (wrapped by meson to capture output)
>             [20/21] Generating pwrmgr_smoketest_scr_vmem64_sim_dv with a custom command
>             [21/21] Generating pwrmgr_smoketest_export_sim_dv with a custom command
>             Building SW image "sw/device/lib/testing/test_rom/test_rom".
>             Building "//sw/device/lib/testing/test_rom" with bazel on network-connected machine.
>             /bin/bash: line 24: bazel: command not found
>             /azp/agent/_work/1/s/hw/dv/tools/dvsim/sim.mk:58: recipe for target 'sw_build' failed
>             make: *** [sw_build] Error 127
> ```

Thanks Cindy!

File: hw/dv/tools/dvsim/sim.mk [hunk 1]
```diff
@@ -88,8 +88,27 @@ ifneq (${sw_images},)
...
+					./bazelisk.sh build //$$(dirname $$image); \
+				else \
+					echo "Building \"//$$(dirname $$image)\" with bazel on air-gapped machine."; \
+					bazel build \
+						--distdir=$$BAZEL_DISTDIR \
+						--repository_cache=$$BAZEL_CACHE \
+						//$$(dirname $$image); \
+				fi; \
+				image_basename="$$(basename $${target}).scr.39.vmem"; \
+				bazel_image_path=$$(find -L $$(bazel info output_path)/ -type f -name "$${image_basename}"); \
```
Comment (cfrantz) [inline hw/dv/tools/dvsim/sim.mk:104]: Above you use `bazelisk.sh` on a network-connected machine, but here you're using `bazel`.  If the environment doesn't know where `bazel` is above, how does it know here?

**Review by cfrantz — APPROVED** [review]
One comment/question.  Otherwise, LGTM.

File: hw/dv/tools/dvsim/sim.mk (hunk 1 above)
Comment (timothytrippel) [reply hw/dv/tools/dvsim/sim.mk:104]: On the air-gapped machine, it is required that the appropriate bazel binary is pre-installed on the machine, and its location on the system is added to the `PATH` environment variable. This is detailed in my "Bazel Air-gapped Builds" document.

Comment (timothytrippel) [reply hw/dv/tools/dvsim/sim.mk:104]: This is failing (private) CI do to a different reason. In fact, the private CI machines are connected to the network, so they make use of the `bazelisk.sh` script. The private CI tests that are failing here, are actually passing when run on the air-gapped nightly regression machine.

Comment (timothytrippel) [reply hw/dv/tools/dvsim/sim.mk:104]: Oh sorry, I just realized what you were referring to, thats a good catch! This actually may impact why this is failing, private CI. Let me fix this and see if it helps.

Comment (timothytrippel) [reply hw/dv/tools/dvsim/sim.mk]: That did the trick. Thanks @cfrantz !

**Pull request merged**